ui/app/mirrors/edit/[mirrorId]/cdc.tsx

```diff
@@ -87,35 +95,112 @@ function summarizeTableClone(clone: QRepMirrorStatus): TableCloneSummary {
 type SnapshotStatusProps = {
   status: SnapshotStatus;
 };
+
+const ROWS_PER_PAGE = 6;
 export const SnapshotStatusTable = ({ status }: SnapshotStatusProps) => {
-  const [snapshotRows, setSnapshotRows] = useState(
-    status.clones.map(summarizeTableClone)
+  const allRows = status.clones.map(summarizeTableClone);
+  const [sortField, setSortField] = useState<
+    'cloneStartTime' | 'avgTimePerPartition'
+  >('cloneStartTime');
+  const [currentPage, setCurrentPage] = useState(1);
+  const totalPages = Math.ceil(allRows.length / ROWS_PER_PAGE);
+
+  const startRow = (currentPage - 1) * ROWS_PER_PAGE;
+  const endRow = startRow + ROWS_PER_PAGE;
+  const [displayedRows, setDisplayedRows] = useState(
```

Comment:
This seems like it should be handled with [`useMemo`](https://react.dev/reference/react/useMemo) based on startRow/endRow/sortField

Reply:
`setDisplayedRows` is required by the SearchBar component so unfortunately right now I can't refactor to `useMemo`.
I can, in a separate PR, bake the search logic into the `useMemo` filter function everywhere in place of the SearchBar component (which will then be removed)

Reply:
Yes, can get that in before this PR